.github/workflows/kind-e2e.yaml

```diff
@@ -192,6 +192,10 @@ jobs:
         # Be KinD to these tests.
         kubectl scale -nknative-serving deployment/chaosduck --replicas=0
 
+        # Setting deadline progress to a shorter value.
+        kubectl patch cm "config-deployment" -n knative-serving \
+          -p '{"data":{"progressDeadline":"120s"}}'
```

Comment:
I don't like this because it means that the SUT you need to pass e2e tests needs more than `test/config`

Reply:
Our current state is a mess because we have patches in `e2e-common`, `e2e-*-test` & `test/config`.

I'm just trying to get the bare minimum to pass prior to refactoring.

Reply:
Selfishly, I'm able to run the e2e tests for ~7 upstreams here without `kubectl patch`ing the SUT: https://github.com/mattmoor/mink/blob/master/.github/workflows/minkind.yaml

I can deal with this failing until we have a better fix, but if it's adding a label to exclude `config-deployment` from `test/config` in the upgrade tests that means bringing back the file you're deleting below so I'm not sure why this is a useful first step?

Do you have a better picture of what things will look like when the refactoring you mention is done?

Reply:
> Selfishly, I'm able to run the e2e tests for ~7 upstreams here without kubectl patching the SUT:

You're not running the entire suite - some features requires changing feature flags etc

Secondly, we have other patches in our kind e2e - so unsure why this is so controversial 

https://github.com/knative/serving/blob/9478d4eac43ca0a3f8da25739e46a8f4b2b13ebd/.github/workflows/kind-e2e.yaml#L186-L190

> I can deal with this failing until we have a better fix, but if it's adding a label to exclude config-deployment from test/config in the upgrade tests that means bringing back the file you're deleting below so I'm not sure why this is a useful first step?
>
> Do you have a better picture of what things will look like when the refactoring you mention is done?

I want to unblock feature work while productivity sorts out their strategy to replace these scripts with something more robust. I don't know their ETA on that

Reply:
What's controversial (to me at least) is that if I have a working Serving setup, today the only real pre-req to run (the bulk of) e2e's is `test/config`.  I consider part of "having a working setup" to include the ingress class annotation you mention.  For alpha/optional tests that require changing the SUT, they are in different suites (e.g. test/e2e/feature).

This patch exists because @vagababov didn't want to wait for the full PDE when running e2e tests, but I'm beginning to regret agreeing to that.  We should remove the patch altogether, make the test run for the duration (it's long 🤷 ), and perhaps skip it if `t.Short()` is true.

Reply:
Hm, 600 is way toooooo long. :)

Reply:
So pass `-short` 🤷

Reply:
We haven't defined or have guidelines what an e2e test is. If I were to look at at `e2e-tests.sh` then it would include scaling, ha, and gated features tests. If I were to look at the GitHub action it's just the tests in `test/e2e`. 

In order for the larger suite to pass we have patches littered in our scripts to setup the system. 
https://github.com/knative/serving/blob/6dcd8a6263b420650a1e47ed1514bd3fa8be79e1/test/e2e-common.sh#L313-L340

They're not in `test/config` already probably because patches can't be `kubectl apply`'d easily. 

To my point earlier - let's fix upgrade tests so they actually work properly (by merging this PR) and then find a solution that addresses the problems you brought up. That to me would be to figure out a way to move patches to `test/config` and provide some way to filter out different patches/configs depending on the test/environment.

Reply:
It feels like we're using a mess in Prow to justify creating a mess in Actions, and it seems there is no owner or an ETA to clean up either mess.

What I suggested devolves to something like
```yaml
metadata:
  name: config-deployment
  # Comment describing why we exclude this during upgrade testing.
  labels:
    exclude-upgrade: "true"
```

... and changing the upgrade test to:
```shell
ko apply -f test/config -l exclude-upgrade!=true
```

Perhaps this can be done via:
```shell
KOFLAGS="-l exclude-upgrade!=true"
```

Reply:
> It feels like we're using a mess in Prow to justify creating a mess in Actions, and it seems there is no owner or an ETA to clean up either mess.

The e2e github action's test surface is simplified and hence never had to deal with this. I can tackle this next if that's your concern.

The solution of setting the `KOFLAGS` seems like a round about way of doing the same thing. For example if I added a Github Action to test the upgrade I'd have to modify include that modified `KOFLAGS` env var.

Reply:
> seems like a round about way of doing the same thing 

I'm trying to optimize for the common case: make running the core e2e tests simple.  I'd rather have the e2e tests take longer and be simple to run than hard to run correctly.

It's not really a hill worth dying on, but its debt that will compound if we don't address it (like the heaping pile of bash).

/unhold